### ISSUE TITLE
Sticking content to the bottom of the page doesn't work

### ISSUE THREAD

**Issue by qJake** [imakewebthings/waypoints#445]
*Regarding shortcut Sticky...*

If you specify that an element (whose natural position is below the viewport on the page somewhere near the bottom) should stick to the bottom of the page using:

```js
new Waypoint.Sticky({
  element: $('.my-section')[0],
  direction: 'up', // or direction: 'up left',
  stuckClass: 'stuck-bottom'
});
```

```css
.stuck-bottom
{
  position: fixed;
  bottom: 0;
  width: 100%;
}
```

When you first load the page, the element is not given the `.stuck-bottom` class, and in fact is not "stuck" until after it reaches the top of the viewport, which is incorrect.

The documentation for the `direction` option led me to believe this was a supported scenario (sticking something to the bottom of the page), and few other sticky JS frameworks support sticking something to the bottom, so I'm hoping this is a bug and not an unsupported use case.

**Edit:** Here's a [JSfiddle demonstrating the issue](http://jsfiddle.net/zqf1ffss/) (the Waypoint code is included inline since I couldn't find the CDN link within a few minutes of looking).

**Comment by stehefan** [comment]
That's because that's not what the Sticky-Shortcut was made for as it mentions in the [documentation](http://imakewebthings.com/waypoints/shortcuts/sticky-elements/): 

*The Sticky Elements shortcut is used to make an element "stick" to the top of the page once a user has scrolled past it.*

You will need to use the normal waypoints-class and use the ['bottom-in-view'](http://imakewebthings.com/waypoints/api/offset-option/#bottom-in-view)-offset.

**Comment by imakewebthings** [comment]
That's *not totally* true Stefan. I'll elaborate more tonight when I'm free. 



—

**Comment by imakewebthings** [comment]
From those docs that @stehefan mention, on the `direction` option:

> In the rare case that you wish to start in a stuck state and lose fixed positioning as you scroll down to the element's natural position, this could be changed to 'up' (or 'left' for the horizontal analog.)

The direction option **is** there to facilitate your use case, @qJake. But the role it plays isn't as plug-and-play as you're expecting. Its value only indicates in which direction the class change happens. By default the stuck class is added on the down/right direction and removed on up/left. If you change the value as @qJake has done, the class is removed on down/right and added on up/left.

This **doesn't** do the other things necessary to achieve your goal:

1. Set the `offset` option to `"bottom-in-view"`, as @stehefan mentioned. (Add that option to the options object.)
2. Give the element that stuck class when the Waypoint is instantiated. (Add this class to the HTML rendered on page load or add it to the element in whatever JS ways you please prior to instantiating the Sticky.)

**Comment by stehefan** [comment]
You learn something new everyday, nice. Thanks for the clarification.

**Comment by qJake** [comment]
Thanks for the explanation, @imakewebthings. That worked for me, with one caveat...

I had to do this in order to get the correct trigger point:

```js
wp.waypoint.triggerPoint += $('.my-section').height();
```

Not sure if that's a bug or not, but it was incorrectly calculating the trigger point for my element, so it was jumping around a bit when the viewport would scroll near that area of the page. The element I stuck to the bottom has a variable height of 150-500px, but it's defined server-side and doesn't change client-side. Not sure why I had to manually adjust the trigger point... possibly something worth looking into?
